### PR TITLE
fix(security): #MVS-108 add new resourceFilter on route get /groupe/enseignement/users/:groupId in GroupeEnseignementController

### PR DESCRIPTION
**Pull request by Gregprod33**
## Describe your changes
- Create a new resource filter which checks the structure from the groupId in parameter
- Change the position of DB.getInstance().init(Neo4j.getInstance(), Sql.getInstance(), MongoDb.getInstance()) in Viescolaire.java
in order to access the instance of neo4j inside resourceProvider;
- Remove some sensitive informations of user in DefaultGroupService.
- Remove unused API (@Get("/groupe/enseignement/user") and @Get("/class/group/:groupId")).

## Checklist tests
Only one route to test :
- **@Get("/groupe/enseignement/users/:groupId")** => competences => saisies projets, then choose a period and a class to activate route
## Issue ticket number and link
[#MVS-108](https://jira.support-ent.fr/browse/MVS-108)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by aymericdevriesere — APPROVED** [review]

File: src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            handler.handle(false);
+            return;
+        }
+        String query = "MATCH(s:Structure)<-[:DEPENDS]-(g:Group) WHERE g.id = {groupId} RETURN DISTINCT s.id as structureId";
+        JsonObject params = new JsonObject();
+
+        params.put(Field.GROUP_ID_CAMEL, groupId);
+
+        neo4j.execute(query ,params, Neo4jResult.validResultHandler(either -> {
+            if (either.right().getValue() != null && !either.right().getValue().isEmpty()) {
```
Comment (JinDouang) [inline src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java:28]: Tu risques d'avoir un NPE si tu check directement l'objet `right`
Pourquoi ne pas faire une vérif classique ? 

`
if (event.isLeft()) {
 // traitement en cas d'erreur
} else {
 // traitement en cas de réussite
}
`

File: src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import fr.wseduc.webutils.http.Binding;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.JsonObject;
+import org.entcore.common.http.filter.ResourcesProvider;
+import org.entcore.common.neo4j.Neo4jResult;
+import org.entcore.common.user.UserInfos;
+
+
+public class StructureAdminPersonnalTeacherFromGroup extends DBService implements ResourcesProvider {
```
Comment (JinDouang) [inline src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java:14]: Tu n'as pas besoin de DBService

File: src/main/java/fr/openent/Viescolaire.java [hunk 3]
```diff
@@ -91,6 +91,7 @@ public class Viescolaire extends BaseServer {
 
 	@Override
     public void start() throws Exception {
+        DB.getInstance().init(Neo4j.getInstance(), Sql.getInstance(), MongoDb.getInstance());
```
Comment (JinDouang) [inline src/main/java/fr/openent/Viescolaire.java:94]: Tu n'as pas besoin de le déplacer, il y a d'autres alternative (je comprends le besoin qui est d'adapter les TUs mais il y a d'autres alternatives)

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Comment by lapeyronnie** [comment]
Peux-tu ajouter, stp, les tests à effectuer pour tester la PR.

File: src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java (hunk 1 above)
Comment (Gregprod33) [reply src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java]: Méthode changée en Future

File: src/main/java/fr/openent/Viescolaire.java (hunk 3 above)
Comment (Gregprod33) [reply src/main/java/fr/openent/Viescolaire.java]: Je l'ai remis à sa place

File: src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+
+    public Future<JsonObject> getStructureByGroup(String groupId){
+        Promise<JsonObject> promise = Promise.promise();
+        String query = "MATCH(s:Structure)<-[:DEPENDS]-(g:Group) WHERE g.id = {groupId} RETURN DISTINCT s.id as structureId";
+        JsonObject params = new JsonObject();
+
+        params.put(Field.GROUP_ID_CAMEL, groupId);
+
+        Neo4j.getInstance().execute(query ,params, Neo4jResult.validUniqueResultHandler(either -> {
```
[the remark this answers is not in the document]
Comment (Gregprod33) [reply src/main/java/fr/openent/viescolaire/security/StructureAdminPersonnalTeacherFromGroup.java]: C'est refacto

**Review by JinDouang — APPROVED** [review]

**Pull request merged**